backend/routes/run.js

```diff
@@ -0,0 +1,194 @@
+import { PrismaClient } from "@prisma/client";
+import {
+  STAT_MEANS,
+  STAT_VARIANCES,
+  POP_SIZE,
+  incrementPopSize,
+} from "./statDictionaries.js";
+const prisma = new PrismaClient();
+
+async function updatePlayer(player) {
+  const oldPlayer = await prisma.player.findUnique({
+    where: { id: player.id },
+  });
+
+  if (oldPlayer == null) {
+    player = await createPlayer(player);
+  } else {
+    player = await changePlayerData(player, oldPlayer);
+  }
+  updatePopulationStats(player);
+}
+
+function updatePopulationStats(player) {
+  if (POP_SIZE === 0) {
```

Comment:
I think you can simplify this function, let me know if I'm misunderstanding something:

1. It looks like you're recalculating all of the values every time a player is updated/created. But really we only need to  calculate this once right? It can be outside of your for loop in `run()`
2. In this way you don't need to keep track of `POP_SIZE` since you'll have access to the entire array of players - you can just sum all of their stats and then divide by the length of the array for the mean.
3. Also it looks like there's no mutations saved to the db what do we do with these stats after calculating them?

Reply:
1. Good point I'd need to keep a running total, but the calculating of the mean should be done in one shot definitely.
2. Yup, and I wouldn't have to worry about dividing by zeros
3. They are global vars that I use later to calculate normalized stats for performance score calculations. Right now they're just stored locally, but I could post it to the Player db table as a Player with id=0, which would also then solve the issue of using global vars at all.